packages/agreement-process/test/vitestSetup.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  StoredEvent,
+  readLastEventByStreamId,
+  setupTestContainersVitest,
+  writeInEventstore,
+  writeInReadmodel,
+  ReadEvent,
+} from "pagopa-interop-commons-test";
+import { inject, afterEach } from "vitest";
+import {
+  Agreement,
+  AgreementEvent,
+  AgreementId,
+  EService,
+  Tenant,
+  toReadModelEService,
+  toReadModelAgreement,
+} from "pagopa-interop-models";
+import { agreementServiceBuilder } from "../src/services/agreementService.js";
+import { agreementQueryBuilder } from "../src/services/readmodel/agreementQuery.js";
+import { attributeQueryBuilder } from "../src/services/readmodel/attributeQuery.js";
+import { eserviceQueryBuilder } from "../src/services/readmodel/eserviceQuery.js";
+import { readModelServiceBuilder } from "../src/services/readmodel/readModelService.js";
+import { tenantQueryBuilder } from "../src/services/readmodel/tenantQuery.js";
+import { toAgreementV1 } from "../src/model/domain/toEvent.js";
+
+export const { readModelRepository, postgresDB, fileManager, cleanup } =
+  setupTestContainersVitest(inject("config"));
+
+afterEach(cleanup);
+
+export const agreements = readModelRepository.agreements;
+export const eservices = readModelRepository.eservices;
+export const tenants = readModelRepository.tenants;
+
+export const readModelService = readModelServiceBuilder(readModelRepository);
+
+const eserviceQuery = eserviceQueryBuilder(readModelService);
+const agreementQuery = agreementQueryBuilder(readModelService);
+const tenantQuery = tenantQueryBuilder(readModelService);
+const attributeQuery = attributeQueryBuilder(readModelService);
+
+export const agreementService = agreementServiceBuilder(
+  postgresDB,
+  agreementQuery,
+  tenantQuery,
+  eserviceQuery,
+  attributeQuery,
+  fileManager
+);
```

Comment:
If I understand correctly, these are going to be instantiated multiple times, once per each test file. Is this a desired behavior?

Reply:
Yes it is, their instantiation have no side effects, so it's fine